### PR TITLE
Unpriv fixups

### PR DESCRIPTION
**Pull request by tych0**
Hey guys,

I'm doing something slightly weird, running lxc inside of lxc-usernsexec. I uncovered some bugs. Take a look.

### CODE REVIEW

File: src/lxc/utils.h [hunk 1]
```diff
@@ -427,8 +427,34 @@ extern int lxc_strmunmap(void *addr, size_t length);
 /* initialize rand with urandom */
 extern int randseed(bool);
 
-inline static bool am_unpriv(void) {
-	return geteuid() != 0;
+inline static bool am_unpriv(void)
```
Comment (hallyn) [inline src/lxc/utils.h]: Can we call this "am_host_unpriv()" or something, to disambiguate from "privileged in a user namespace" ?

**Comment by hallyn** [comment]
Thanks - LGTM.  @brauner any objections?

File: src/lxc/utils.h (hunk 1 above)
Comment (tych0) [reply src/lxc/utils.h]: Sure, I'll push that change shortly.

**Comment by brauner** [comment]
Fine with me that's something I had planned on doing before and was pinged on by some runc guys. Excellent timing! :)

**Pull request merged**